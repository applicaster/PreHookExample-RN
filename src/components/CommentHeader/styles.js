import {
  Dimensions,
  Platform,
  StyleSheet,
} from 'react-native';

const iOS = 'ios';
const width = Dimensions.get('window').width;
export const styles = StyleSheet.create({
  commentHeader: {
    flexDirection: 'row',
    paddingHorizontal: 12,
    paddingTop: 12,
  },
  commentHeaderInfo: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    paddingLeft: 10,
    height: 43,
    width,
  },
  commentHeaderInfoWithoutUserName: {
    justifyContent: 'center',
  },
  avatar: {
    height: 40,
    width: 40,
  },
  name: {
    fontSize: 14,
    marginTop: (Platform.OS === iOS) ? -3 : -4,
    marginBottom: -2,
    fontWeight: '600',
  },
  userName: {
    fontSize: 10,
  },
});
