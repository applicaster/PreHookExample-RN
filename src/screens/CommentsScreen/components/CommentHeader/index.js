import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Image,
  Text,
  View,
} from 'react-native';
import EventTimestamp from '../EventTimestamp';
import { styles } from './styles';

class CommentHeader extends Component {
  renderUserName() {
    const { userName } = this.props;
    const colorStyle = { color: this.context.textColor };
    
    return (userName) ? <Text style={[styles.userName, colorStyle]} numberOfLines={1}>@{userName}</Text> : null;
  }

  renderName() {
    const { name } = this.props;
    const colorStyle = { color: this.context.mainColor };

    return <Text style={[styles.name, colorStyle]} numberOfLines={1}>{name}</Text>;
  }

  renderTimestamp() {
    const { createdAt } = this.props;
    const colorStyle = { color: this.context.textColor };

    return <EventTimestamp timestamp={createdAt} colorStyle={colorStyle} />;
  }

  renderHeaderInfo() {
    const { userName } = this.props;
    return (
      <View style={[styles.commentHeaderInfo, !userName && styles.commentHeaderInfoWithoutUserName]}>
        <View style={styles.userInfoContainer}>
          {this.renderName()}
          {this.renderUserName()}
        </View>
        <View style={styles.timestampContainer}>
          {this.renderTimestamp()}
        </View>
      </View>
    );
  }

  renderAvatarImage() {
    const { avatarImageUrl } = this.props;
    return <Image style={styles.avatar} source={{ uri: avatarImageUrl }} />;
  }

  render() {
    return (
      <View style={[styles.commentHeader]}>
        {this.renderAvatarImage()}
        {this.renderHeaderInfo()}
      </View>
    );
  }
}

CommentHeader.propTypes = {
  avatarImageUrl: PropTypes.string,
  createdAt: PropTypes.number,
  name: PropTypes.string,
  userName: PropTypes.string,
};

CommentHeader.contextTypes = {
  mainColor: PropTypes.string,
  textColor: PropTypes.string,
};

export default CommentHeader;
