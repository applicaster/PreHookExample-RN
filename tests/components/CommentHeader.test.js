import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CommentHeader from '../../src/components/CommentHeader';
import { comment } from './testDataUtils';

describe('CommentHeader', () => {
  test('component renders correctly', () => {
    const someComment = comment(1);
    const html = renderer.create(
      <CommentHeader
        avatarImageUrl={someComment.user.avatarImageUrl}
        createdAt={someComment.createdAt}
        name={someComment.name}
        userName={someComment.user.userName}
      />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly for Twitter or Instagram comment', () => {
    const someComment = comment(1, 'image', 'twitter', 'applicaster');
    const html = renderer.create(
      <CommentHeader
        avatarImageUrl={someComment.user.avatarImageUrl}
        createdAt={someComment.createdAt}
        name={someComment.name}
        userName={someComment.user.userName}
      />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly for Facebook comment', () => {
    const someComment = comment(1, 'image', 'facebook', null);
    const html = renderer.create(
      <CommentHeader
        avatarImageUrl={someComment.user.avatarImageUrl}
        createdAt={someComment.createdAt}
        name={someComment.name}
        userName={someComment.user.userName}
      />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
