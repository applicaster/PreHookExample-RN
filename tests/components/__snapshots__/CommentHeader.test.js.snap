// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentHeader component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "paddingHorizontal": 12,
        "paddingTop": 12,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://scontent.cdninstagram.com/t51.2885-19/11356956_452315591560419_1118499770_a.jpg",
      }
    }
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
          "height": 43,
          "justifyContent": "space-between",
          "paddingLeft": 10,
          "width": 750,
        },
        false,
      ]
    }
  >
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 14,
              "fontWeight": "600",
              "marginBottom": -2,
              "marginTop": -3,
            },
            Object {
              "color": undefined,
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 10,
            },
            Object {
              "color": undefined,
            },
          ]
        }
      >
        @
        applicaster
      </Text>
    </View>
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 12,
            },
            Object {
              "color": undefined,
            },
          ]
        }
      >
        August 4
      </Text>
    </View>
  </View>
</View>
`;

exports[`CommentHeader component renders correctly for Facebook comment 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "paddingHorizontal": 12,
        "paddingTop": 12,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://scontent.cdninstagram.com/t51.2885-19/11356956_452315591560419_1118499770_a.jpg",
      }
    }
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
          "height": 43,
          "justifyContent": "space-between",
          "paddingLeft": 10,
          "width": 750,
        },
        false,
      ]
    }
  >
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 14,
              "fontWeight": "600",
              "marginBottom": -2,
              "marginTop": -3,
            },
            Object {
              "color": undefined,
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 10,
            },
            Object {
              "color": undefined,
            },
          ]
        }
      >
        @
        facebook
      </Text>
    </View>
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 12,
            },
            Object {
              "color": undefined,
            },
          ]
        }
      >
        August 4
      </Text>
    </View>
  </View>
</View>
`;

exports[`CommentHeader component renders correctly for Twitter or Instagram comment 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "paddingHorizontal": 12,
        "paddingTop": 12,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://scontent.cdninstagram.com/t51.2885-19/11356956_452315591560419_1118499770_a.jpg",
      }
    }
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
          "height": 43,
          "justifyContent": "space-between",
          "paddingLeft": 10,
          "width": 750,
        },
        false,
      ]
    }
  >
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 14,
              "fontWeight": "600",
              "marginBottom": -2,
              "marginTop": -3,
            },
            Object {
              "color": undefined,
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 10,
            },
            Object {
              "color": undefined,
            },
          ]
        }
      >
        @
        twitter
      </Text>
    </View>
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 12,
            },
            Object {
              "color": undefined,
            },
          ]
        }
      >
        August 4
      </Text>
    </View>
  </View>
</View>
`;
